eqcorrscan/core/match_filter/tribe.py

```diff
@@ -586,55 +729,389 @@ def detect(self, stream, threshold, threshold_type, trig_int, plot=False,
+                elif isinstance(to_corr, Poison):
+                    Logger.info("Killed in main loop")
+                    break
+                starttime, i, stream, template_names, templates,\
+                    *extras = to_corr
+                inner_kwargs = copy.copy(kwargs)  # We will mangle them
+                # Correlation specific handling to reduce single-threaded time
+                if xcorr_func == "fmf":
+                    weights, pads, chans, no_chans = extras
+                    inner_kwargs.update({
+                        'weights': weights, 'pads': pads, "no_chans": no_chans,
+                        "chans": chans, "prepped": True})
+                elif xcorr_func in (None, 'fftw'):
+                    pads, seed_ids = extras
+                    inner_kwargs.update({
+                        "pads": pads, "seed_ids": seed_ids, "prepped": True})
+
+                Logger.info(f"Got stream of {len(stream)} channels")
+                Logger.info(f"Starting correlation from {starttime}")
+
+                all_peaks, thresholds, no_chans, chans = _corr_and_peaks(
+                    templates=templates, template_names=template_names,
+                    stream=stream, xcorr_func=xcorr_func,
+                    concurrency=concurrency,
+                    cores=cores, i=i, export_cccsums=export_cccsums,
+                    parallel=parallel, peak_cores=peak_cores,
+                    threshold=threshold, threshold_type=threshold_type,
+                    trig_int=trig_int, sampling_rate=sampling_rate,
+                    full_peaks=full_peaks, plot=plot, plotdir=plotdir,
+                    plot_format=plot_format, **inner_kwargs
+                )
+                peaks_queue.put(
+                    (starttime, all_peaks, thresholds, no_chans, chans,
+                     template_names))
+            except Exception as e:
+                Logger.error(
+                    f"Caught exception in correlator:\n {e}")
+                traceback.print_tb(e.__traceback__)
+                poison_queue.put(e)
+                break  # We need to break in Main
+            i += 1
+        Logger.debug("Putting None into peaks queue.")
+        peaks_queue.put(None)
+
+        # Get the party back
+        Logger.info("Collecting party")
+        party = Party()
+        while True:
+            killed = _check_for_poison(poison_queue)
+            if killed:
+                Logger.error("Killed")
+                break
+            pf = _get_and_check(party_file_queue, poison_queue)
+            if pf is None:
+                break
+            if isinstance(pf, Poison):
+                Logger.error("Killed")
+                break
+            with open(pf, "rb") as f:
+                party += pickle.load(f)
+            if not save_progress:
+                os.remove(pf)
+        if not save_progress:
+            if os.path.isdir(".parties"):
+                shutil.rmtree(".parties")
```

Comment:
```suggestion

```

Removing the directory might affect other concurrently running routines (e.g. handled by a different python interpreter).